FoodOrderingApp-service/src/main/java/com/upgrad/FoodOrderingApp/service/businness/RestaurantService.java

```diff
@@ -15,17 +20,23 @@
     @Autowired
     private RestaurantDao restaurantDao;
 
+    @Autowired
+    private CategoryDao categoryDao;
+
+    @Autowired
+    private RestaurantCategoryDao restaurantCategoryDao;
+
     /**
-     * @return List of all restaurants
+     * @return List of all restaurants order by rating
      */
-    public List<RestaurantEntity> getAllRestaurants() {
+    public List<RestaurantEntity> restaurantsByRating() {
         return restaurantDao.getAllRestaurants();
     }
 
     /**
-     * @return List of all restaurants
+     * @return List of all restaurants by entered input order by restaurant name
      */
-    public List<RestaurantEntity> getRestaurantsByName(final String restaurantName)
+    public List<RestaurantEntity> restaurantsByName(final String restaurantName)
     throws RestaurantNotFoundException {
         //If the restaurant name field entered by the customer is empty, throw exception
         if(restaurantName == null || restaurantName == "") {
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [ES: Comparison of String objects using == or != (ES_COMPARING_STRINGS_WITH_EQ)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#es-comparison-of-string-objects-using-or-es-comparing-strings-with-eq).
This code compares java.lang.String objects for reference equality using the == or != operators. Unless both strings are either constants in a source file, or have been interned using the String.intern() method, the same string value may be represented by two different String objects. Consider using the equals(Object) method instead.